usb/esp_tinyusb/include_private/tinyusb_ext_phy.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2020-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "esp_err.h"
+#include "esp_private/usb_phy.h"
+#include "soc/usb_pins.h"
+#include "tinyusb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Configure and install new USB PHY
+ *
+ * @param[in] config tinyusb stack specific configuration
+ */
+esp_err_t tinyusb_ext_phy_new(const tinyusb_config_t *config);
+
+/**
+ * @brief Delete previously installed USB PHY
+ *
+ */
+esp_err_t tinyusb_ext_phy_delete(void);
```

Comment:
@roma-jam Not sure if this is a good idea. We'll end up having two separate PHY APIs (one for host stack, one for device stack).

Is there a reason why we can't use the existing `usb_phy.h` API?

Reply:
@Dazza0 Actually, the only thing that this changes makes is they collect the `usb_phy_*` calls in a one place (`tinyusb_ext_phy.c`) and add the dependency to build or not to build `usb_phy` based on target (S2/S3 or P4).
There is no changes in API, not implementing new API or anything like this.
And we still using existing usb_phy.h API.

Reply:
@roma-jam 

In theory, the P4 should still make a call to `usb_phy.h` API. In fact, USB Host, Device, and USJ should all be making `usb_phy.h` calls on initialization to specify which controller they are, and which PHY they are using.

Future HS capable targets may have multiple possible PHYs (e.g., UTMI vs ULPI), thus we'll need to add a `usb_phy_...` call back in for those targets.

Furthermore, even if the P4 HS only has one possible PHY it can use (i.e., UTMI), it's still good practice to call `usb_phy_new()` even though are no other possible choices for PHY. Because `usb_phy_new()` not only acts as PHY selection, but also PHY initialization. For example, what if before the device stack was initialize, the power management/sleep services ran and turned off the USB PHY clock to save power. Having a `usb_phy_new()` call to turn it back on would be helpful.